### DISCUSSION TITLE
本地开发组件bug

### DISCUSSION THREAD

**Discussion by Burninghamster** [illacloud/illa-builder#786] [Q&A]
Dropdown这个下拉组件肯定有性能问题，因为我每次操作这个组件的时候页面卡死，浏览器不卡死，只要是涉及到这个下拉框的功能页面都会都会卡死

**Comment by AruSeito** [comment]
您好，请问您是在编辑器接入了 dropdown 这个组件吗？还是在dashboard 的点击 moreAction 等也会卡死呢？

**Reply by Burninghamster** [reply]
![1673424072224](https://user-images.githubusercontent.com/33534997/211750605-887092c6-5edd-4464-96a9-9ddee732a5c3.png)我这边环境是win10,  pnpm run            比如删除组件，鼠标右键触发下拉框也会卡死，不过像切换组件，还有弹窗组件就不会出现这种情况，只要是鼠标点击或者右击出现下拉框都是会卡死，我自己在本地测试了，删除这个组件，随便怎么操作都不会出现这个情况，反正只要是使用了dropdown这个组件的地方无论什么功能都会造成这个结果

**Reply by AruSeito** [reply]
您好，感谢您的反馈。我们接下来会进行排查一下看看。

**Reply by Burninghamster** [reply]
我是用的你们的主分支，你下载试试看，一定有问题，组件依赖是有冲突的，分别是react-color和illa-design  dropdown组件，如果下载这个组件illa-design  dropdown 1.0.12版本必须降级到1.0.10，不降级的话，项目整个直接跑不起来的，下载依赖好像需要cnpm npm pnpm yarn的前端命令，这几个配合起来才可以把所有的依赖下完. 然后在pnpm dev跑

**Reply by AruSeito** [reply]
您好，看您的描述应该是在安装依赖的时候出了问题，导致在用 dropdown 的时候出现了问题，建议使用默认源并且使用 pnpm 安装，多个包管理混合着用会出现不可预知问题。

**Reply by Burninghamster** [reply]
您好，我找了5个人测试的，主分支如果直接用pnpm确实有问题，你们比较稳定的是哪个分支，我在看一下？
![1673603562173](https://user-images.githubusercontent.com/33534997/212290870-c6fb6e3f-ac19-4a6f-a44a-93585b9b48b8.png)
![1673603786258](https://user-images.githubusercontent.com/33534997/212291580-55c2bd1b-5db8-4964-b36b-f57e71100a87.png)
这个json功能是在分享里面还是在哪里，还是必须部署到docker环境我才看得到,我在开发调试里面是看不到官网说的这些功能的？？？

**Reply by AruSeito** [reply]
您好，稳定分支就是 main。后面我们会再次用 windows 进行排查一下看一下。

图片中的回复：
您可以通过放置 Icon 组件（未来会加上，目前还没有）然后配置eventHandler 来关闭弹窗，相应的，您可以使用button 在页脚上配置。

第三点的回复：
抱歉给您带来困扰，这些功能都是我们在排期中的，正在抓紧开发中，我们完成之后会第一时间联系您。

**Comment by owenlongbo** [comment]
重构版本已经完成了，可以再看一下。
